### ISSUE TITLE
`bst build` appears to build same module twice

### ISSUE THREAD

**Issue by mcatanzaro** [apache/buildstream#1474]
I noticed bst 1.6.1 is building webkitgtk.bst twice at the same time:

![Screenshot from 2021-05-03 16-18-29](https://user-images.githubusercontent.com/1424966/116935536-1709f880-ac2c-11eb-821a-b48ac16ceb9e.png)

That's twice the WebKit I had bargained for! I'm not certain, but I think it is not just a visual error: I think it really is building WebKit twice at once, because I see two separate chains of bst processes in gnome-control-center, each working on webkitgtk:

![Screenshot from 2021-05-03 16-21-03](https://user-images.githubusercontent.com/1424966/116935559-1ec99d00-ac2c-11eb-9468-4e6482b0e9b2.png)

![Screenshot from 2021-05-03 16-21-07](https://user-images.githubusercontent.com/1424966/116935565-1f623380-ac2c-11eb-9ab1-65b5f63774ad.png)

I attached two screenshots of that to show the tooltips for the two different bwrap processes that you can see are both working on webkitgtk.bst. Another clue that it's really doing a second build: my desktop was thrashing, which doesn't normally happen when building WebKitGTK via buildstream.

This is, sadly, not reproducible at all. After killing buildstream with Ctrl+C and restarting the build, it works fine, building webkitgtk.bst only once.

**Comment by mcatanzaro** [comment]
Abderrahim says: "I noticed this can happen if fetching the sources fails and is retried"

And that's probably exactly what happened! Something is currently wonky with DNS for webkitgtk.org, name resolution often fails and then succeeds again on the next try....

**Comment by nanonyme** [comment]
It seems to me this same problem still exists in bst2. And it can also happen in non-interactive mode resulting in CI building same things twice in parallel.

**Comment by abderrahim** [comment]
Yup, I noticed it with bst2 too (with manual retry). Don't know if it's the same issue happening in non-interactive mode.
